From 657383367722fede3d486c8733706cb6586eb3a3 Mon Sep 17 00:00:00 2001
From: Jianqun Xu <jay.xu@rock-chips.com>
Date: Tue, 13 Dec 2016 17:29:34 +0800
Subject: [PATCH] ARM64: dts: rockchip: move rk3399 pmu-io-domain nodes to the
 pmugrf

Afer the "PM / AVS: rockchip-io: make io-domains a child of the GRF",
the pmu-io-domains should be a sub-node of the pmugrf simple-mfd.

Change-Id: Iebfe9041a604580ce3e5b028d7a143fcdbbdff25
Signed-off-by: Jianqun Xu <jay.xu@rock-chips.com>
---
 .../boot/dts/rockchip/rk3399-box-808-android.dts    | 12 +++++-------
 arch/arm64/boot/dts/rockchip/rk3399-box.dtsi        | 13 ++++++-------
 arch/arm64/boot/dts/rockchip/rk3399-evb.dtsi        | 13 ++++++-------
 arch/arm64/boot/dts/rockchip/rk3399-gru.dtsi        | 12 +++++-------
 .../dts/rockchip/rk3399-mid-818-android-next.dts    | 12 +++++-------
 .../boot/dts/rockchip/rk3399-mid-818-android.dts    | 13 ++++++-------
 arch/arm64/boot/dts/rockchip/rk3399-rv1-android.dts | 11 +++++------
 arch/arm64/boot/dts/rockchip/rk3399-sapphire.dtsi   | 11 +++++------
 arch/arm64/boot/dts/rockchip/rk3399-tve1205g.dts    | 12 +++++-------
 arch/arm64/boot/dts/rockchip/rk3399-vr-android.dts  | 13 ++++++-------
 arch/arm64/boot/dts/rockchip/rk3399.dtsi            |  5 +++++
 11 files changed, 59 insertions(+), 68 deletions(-)

diff --git a/arch/arm64/boot/dts/rockchip/rk3399-box-808-android.dts b/arch/arm64/boot/dts/rockchip/rk3399-box-808-android.dts
index ac59eb9df090..dd68b23cc642 100644
--- a/arch/arm64/boot/dts/rockchip/rk3399-box-808-android.dts
+++ b/arch/arm64/boot/dts/rockchip/rk3399-box-808-android.dts
@@ -115,13 +115,6 @@
 		#clock-cells = <0>;
 	};
 
-	pmu-io-domains {
-		compatible = "rockchip,rk3399-pmu-io-voltage-domain";
-		rockchip,grf = <&pmugrf>;
-
-		pmu1830-supply = <&vcc_1v8>;
-	};
-
 	spdif-sound {
 		status = "okay";
 		compatible = "simple-audio-card";
@@ -793,3 +786,8 @@
 	status = "okay";
 };
 
+&pmu_io_domains {
+	status = "okay";
+	pmu1830-supply = <&vcc_1v8>;
+};
+
diff --git a/arch/arm64/boot/dts/rockchip/rk3399-box.dtsi b/arch/arm64/boot/dts/rockchip/rk3399-box.dtsi
index cc72bf15aac5..9014d1c29b26 100644
--- a/arch/arm64/boot/dts/rockchip/rk3399-box.dtsi
+++ b/arch/arm64/boot/dts/rockchip/rk3399-box.dtsi
@@ -112,13 +112,6 @@
 		#clock-cells = <0>;
 	};
 
-	pmu-io-domains {
-		compatible = "rockchip,rk3399-pmu-io-voltage-domain";
-		rockchip,grf = <&pmugrf>;
-
-		pmu1830-supply = <&vcc_1v8>;
-	};
-
 	spdif-sound {
 		status = "okay";
 		compatible = "simple-audio-card";
@@ -860,3 +853,9 @@
 &pmu_pvtm {
 	status = "okay";
 };
+
+&pmu_io_domains {
+	status = "okay";
+	pmu1830-supply = <&vcc_1v8>;
+};
+
diff --git a/arch/arm64/boot/dts/rockchip/rk3399-evb.dtsi b/arch/arm64/boot/dts/rockchip/rk3399-evb.dtsi
index 16f02111dc36..696004f351b3 100644
--- a/arch/arm64/boot/dts/rockchip/rk3399-evb.dtsi
+++ b/arch/arm64/boot/dts/rockchip/rk3399-evb.dtsi
@@ -118,13 +118,6 @@
 		regulator-boot-on;
 	};
 
-	pmu-io-domains {
-		compatible = "rockchip,rk3399-pmu-io-voltage-domain";
-		rockchip,grf = <&pmugrf>;
-
-		pmu1830-supply = <&vcc1v8_pmu>;
-	};
-
 	es8316-sound {
 		compatible = "simple-audio-card";
 		simple-audio-card,format = "i2s";
@@ -576,3 +569,9 @@
 &pmu_pvtm {
 	status = "okay";
 };
+
+&pmu_io_domains {
+	status = "okay";
+	pmu1830-supply = <&vcc1v8_pmu>;
+};
+
diff --git a/arch/arm64/boot/dts/rockchip/rk3399-gru.dtsi b/arch/arm64/boot/dts/rockchip/rk3399-gru.dtsi
index e2b5c11f4ee2..4fb9acab61a5 100644
--- a/arch/arm64/boot/dts/rockchip/rk3399-gru.dtsi
+++ b/arch/arm64/boot/dts/rockchip/rk3399-gru.dtsi
@@ -464,13 +464,6 @@
 
 	/* END REGULATORS */
 
-	pmu-io-domains {
-		compatible = "rockchip,rk3399-pmu-io-voltage-domain";
-		rockchip,grf = <&pmugrf>;
-
-		pmu1830-supply = <&pp1800_pmu>;		/* PMUIO2_VDD */
-	};
-
 	sound {
 		compatible = "rockchip,rockchip-audio-da7219";
 		rockchip,model = "audio-da7219";
@@ -946,6 +939,11 @@ ap_i2c_audio: &i2c8 {
 	status = "okay";
 };
 
+&pmu_io_domains {
+	status = "okay";
+	pmu1830-supply = <&pp1800_pmu>;		/* PMUIO2_VDD */
+};
+
 /* PINCTRL: always below everything else */
 
 &pinctrl {
diff --git a/arch/arm64/boot/dts/rockchip/rk3399-mid-818-android-next.dts b/arch/arm64/boot/dts/rockchip/rk3399-mid-818-android-next.dts
index 9e5c0348564d..f88705a15b19 100644
--- a/arch/arm64/boot/dts/rockchip/rk3399-mid-818-android-next.dts
+++ b/arch/arm64/boot/dts/rockchip/rk3399-mid-818-android-next.dts
@@ -158,13 +158,6 @@
 		regulator-boot-on;
 	};
 
-	pmu-io-domains {
-		compatible = "rockchip,rk3399-pmu-io-voltage-domain";
-		rockchip,grf = <&pmugrf>;
-
-		pmu1830-supply = <&vcc_1v8>;
-	};
-
 	es8316-sound {
 		compatible = "simple-audio-card";
 		simple-audio-card,format = "i2s";
@@ -1009,6 +1002,11 @@
 	};
 };
 
+&pmu_io_domains {
+	status = "okay";
+	pmu1830-supply = <&vcc_1v8>;
+};
+
 &route_edp {
 	status = "okay";
 };
diff --git a/arch/arm64/boot/dts/rockchip/rk3399-mid-818-android.dts b/arch/arm64/boot/dts/rockchip/rk3399-mid-818-android.dts
index 30c11e5c3f24..8af5d80db980 100644
--- a/arch/arm64/boot/dts/rockchip/rk3399-mid-818-android.dts
+++ b/arch/arm64/boot/dts/rockchip/rk3399-mid-818-android.dts
@@ -146,13 +146,6 @@
 		regulator-boot-on;
 	};
 
-	pmu-io-domains {
-		compatible = "rockchip,rk3399-pmu-io-voltage-domain";
-		rockchip,grf = <&pmugrf>;
-
-		pmu1830-supply = <&vcc_1v8>;
-	};
-
 	es8316-sound {
 		compatible = "simple-audio-card";
 		simple-audio-card,format = "i2s";
@@ -1020,3 +1013,9 @@
 &pmu_pvtm {
 	status = "okay";
 };
+
+&pmu_io_domains {
+	status = "okay";
+	pmu1830-supply = <&vcc_1v8>;
+};
+
diff --git a/arch/arm64/boot/dts/rockchip/rk3399-rv1-android.dts b/arch/arm64/boot/dts/rockchip/rk3399-rv1-android.dts
index 2944593401e8..cc72207d2cea 100644
--- a/arch/arm64/boot/dts/rockchip/rk3399-rv1-android.dts
+++ b/arch/arm64/boot/dts/rockchip/rk3399-rv1-android.dts
@@ -98,12 +98,6 @@
 		};
 	};
 
-	pmu-io-domains {
-		compatible = "rockchip,rk3399-pmu-io-voltage-domain";
-		rockchip,grf = <&pmugrf>;
-		pmu1830-supply = <&vcc_1v8>;
-	};
-
 	dw_hdmi_audio: dw-hdmi-audio {
 		status = "okay";
 		compatible = "rockchip,dw-hdmi-audio";
@@ -683,6 +677,11 @@
 	mali-supply = <&vdd_gpu>;
 };
 
+&pmu_io_domains {
+	status = "okay";
+	pmu1830-supply = <&vcc_1v8>;
+};
+
 &pinctrl {
 	sdio-pwrseq {
 		wifi_enable_h: wifi-enable-h {
diff --git a/arch/arm64/boot/dts/rockchip/rk3399-sapphire.dtsi b/arch/arm64/boot/dts/rockchip/rk3399-sapphire.dtsi
index c070547250d1..2d3bb78576fc 100644
--- a/arch/arm64/boot/dts/rockchip/rk3399-sapphire.dtsi
+++ b/arch/arm64/boot/dts/rockchip/rk3399-sapphire.dtsi
@@ -115,12 +115,6 @@
 		};
 	};
 
-	pmu-io-domains {
-		compatible = "rockchip,rk3399-pmu-io-voltage-domain";
-		rockchip,grf = <&pmugrf>;
-		pmu1830-supply = <&vcc_3v0>;
-	};
-
 	sdio_pwrseq: sdio-pwrseq {
 		compatible = "mmc-pwrseq-simple";
 		clocks = <&rk808 1>;
@@ -512,6 +506,11 @@
 	status = "okay";
 };
 
+&pmu_io_domains {
+	status = "okay";
+	pmu1830-supply = <&vcc_3v0>;
+};
+
 &pwm0 {
 	status = "okay";
 };
diff --git a/arch/arm64/boot/dts/rockchip/rk3399-tve1205g.dts b/arch/arm64/boot/dts/rockchip/rk3399-tve1205g.dts
index c87980ae9f02..f1ba7a06eac8 100644
--- a/arch/arm64/boot/dts/rockchip/rk3399-tve1205g.dts
+++ b/arch/arm64/boot/dts/rockchip/rk3399-tve1205g.dts
@@ -149,13 +149,6 @@
 		regulator-boot-on;
 	};
 
-	pmu-io-domains {
-		compatible = "rockchip,rk3399-pmu-io-voltage-domain";
-		rockchip,grf = <&pmugrf>;
-
-		pmu1830-supply = <&vcc_1v8>;
-	};
-
 	es8316-sound {
 		compatible = "simple-audio-card";
 		simple-audio-card,format = "i2s";
@@ -796,6 +789,11 @@
 	status = "okay";
 };
 
+&pmu_io_domains {
+	status = "okay";
+	pmu1830-supply = <&vcc_1v8>;
+};
+
 &pwm0 {
 	status = "okay";
 };
diff --git a/arch/arm64/boot/dts/rockchip/rk3399-vr-android.dts b/arch/arm64/boot/dts/rockchip/rk3399-vr-android.dts
index ebf80e191fb0..09329e9031af 100644
--- a/arch/arm64/boot/dts/rockchip/rk3399-vr-android.dts
+++ b/arch/arm64/boot/dts/rockchip/rk3399-vr-android.dts
@@ -134,13 +134,6 @@
 		regulator-boot-on;
 	};
 
-	pmu-io-domains {
-		compatible = "rockchip,rk3399-pmu-io-voltage-domain";
-		rockchip,grf = <&pmugrf>;
-
-		pmu1830-supply = <&vcc_1v8>;
-	};
-
 	es8316-sound {
 		compatible = "simple-audio-card";
 		simple-audio-card,format = "i2s";
@@ -971,3 +964,9 @@
 &pmu_pvtm {
 	status = "okay";
 };
+
+&pmu_io_domains {
+	status = "okay";
+	pmu1830-supply = <&vcc_1v8>;
+};
+
diff --git a/arch/arm64/boot/dts/rockchip/rk3399.dtsi b/arch/arm64/boot/dts/rockchip/rk3399.dtsi
index fb1448296be1..f050c7ee4609 100644
--- a/arch/arm64/boot/dts/rockchip/rk3399.dtsi
+++ b/arch/arm64/boot/dts/rockchip/rk3399.dtsi
@@ -1058,6 +1058,11 @@
 		compatible = "rockchip,rk3399-pmugrf", "syscon", "simple-mfd";
 		reg = <0x0 0xff320000 0x0 0x1000>;
 
+		pmu_io_domains: pmu-io-domains {
+			compatible = "rockchip,rk3399-pmu-io-voltage-domain";
+			status = "disabled";
+		};
+
 		reboot-mode {
 			compatible = "syscon-reboot-mode";
 			offset = <0x300>;
-- 
2.35.3

